bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/AnnotationActionHandler.java

```diff
@@ -146,6 +147,8 @@ public AnnotationActionHandler(Action module, ActionType mt, Method method, Obje
                 output.put(MODULE_RESULT, doubleValue);
             } else if (result instanceof Float floatValue) {
                 output.put(MODULE_RESULT, floatValue);
+            } else if (result instanceof BigDecimal bigDecimalValue) {
+                output.put(MODULE_RESULT, bigDecimalValue);
```

Comment:
What about converting to double?

Reply:
Changing the type is not a problem ?
This code is not used when the thing action is called from inside a rule ?

Reply:
Correct.
This code is only called if the Thing action is a action module itself, which is only the case if it is invoked through the Thing actions API or added to a UI rule from the add action menu, where the output isn’t used at the moment and should be a primitive.
But when called from code in a rule, the action is called directly and not through this code here.

Reply:
Changed